core/src/it/scala/org/ensime/core/AnalyzerManagerSpec.scala

```diff
@@ -0,0 +1,126 @@
+import org.ensime.util.file._
+class AnalyzerManagerSpec extends EnsimeSpec
+    with IsolatedAnalyzerManagerFixture {
+  /** The definition of the original project to clone for testing. */
+  val original = EnsimeConfigFixture.SimpleMultiModuleTestProject
+
+  "Analyzer Manager" should "handle multi-module requests correctly" in {
+    withEnsimeConfig { implicit config =>
+      withTestKit { implicit testkit =>
+        withAnalyzerManager { (analyzerManager, asyncHelper) =>
+          import testkit._
+          val firstSourceRoot = scalaMains(config).head
+          val fooFile = firstSourceRoot / "org/example/Foo.scala"
+          val fooPath = fooFile.toPath
+          val fooFilePath = fooFile.getAbsolutePath
+          val barFile = firstSourceRoot / "org/example/Bar.scala"
+          val barPath = barFile.toPath
+          val secondSourceRoot = scalaMains(config)(1)
+          val foo2File = secondSourceRoot / "org/example2/Foo2.scala"
+          val foo2Path = foo2File.getAbsolutePath
+          val bar2File = secondSourceRoot / "org/example2/Bar2.scala"
+          val bar2Path = bar2File.getAbsolutePath
+
+          analyzerManager ! TypecheckFilesReq(List(Left(fooFile), Left(barFile), Left(foo2File), Left(bar2File)))
+          expectMsg(VoidResponse)
+
+          val missingFile = firstSourceRoot / "missing.scala"
+          val missingFilePath = missingFile.getAbsolutePath
+          val anotherMissingFile = secondSourceRoot / "missingToo.scala"
+          val anotherMissingFilePath = anotherMissingFile.getAbsolutePath
+
+          analyzerManager ! TypecheckFilesReq(List(Left(missingFile), Left(anotherMissingFile)))
+          val errorDescription = expectMsgType[EnsimeServerError].description
+          errorDescription should include(s"""file(s): "${EnsimeFile(missingFilePath)}" do not exist""")
+          errorDescription should include(s"""file(s): "${EnsimeFile(anotherMissingFilePath)}" do not exist""")
+
+          analyzerManager ! TypeByNameReq("org.example.Bloo")
+          expectMsgType[api.BasicTypeInfo]
+
+          analyzerManager ! UnloadModuleReq(EnsimeProjectId("testing_simple", "compile"))
+          expectMsg(VoidResponse)
+
+          analyzerManager ! TypeByNameReq("org.example.Bloo")
+          expectMsg(BasicTypeInfo("<none>", DeclaredAs.Nil, "<none>"))
+
+          analyzerManager ! TypecheckFilesReq(List(Left(fooFile), Left(barFile)))
+          expectMsg(VoidResponse)
+
+          asyncHelper.fishForMessage() {
+            case FullTypeCheckCompleteEvent => true
+            case _ => false
+          }
+
+          analyzerManager ! InspectPackageByPathReq("org.example")
+          val packageInfo = expectMsgType[PackageInfo]
+          packageInfo.name shouldBe "example"
+          packageInfo.fullName shouldBe "org.example"
+
+          packageInfo.members.collect {
+            case b: api.BasicTypeInfo => b.copy(pos = None)
+          } should contain theSameElementsAs List(
```

Comment:
I'm not sure the exact contents matter... it's enough to assert that the list is non-empty. The exact details can change between scala versions so over-specifying tests is often be a burden.